crates/conversions/src/byte_array.rs

```diff
@@ -68,3 +70,32 @@ impl From<ByteArray> for String {
         format!("{full_words_string}{pending_word_string}")
     }
 }
+
+
+impl fmt::Display for ByteArray {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{self:?}") 
```

Comment:
In the `Display` implementation, we want to display the original string.

I would remove the `From<ByteArray> for String` conversion and integrate its code into the `Display` implementation.